### PR TITLE
chat feature update.

### PR DESCRIPTION
**Pull request by lyw0149**
1. UI update for V2.
2. push notification enabled.
3. Improved app stability.

### CODE REVIEW

File: src/components/Activity/Activity.js [hunk 1]
```diff
@@ -26,13 +26,21 @@ class Activity extends Component {
   componentDidMount() {
     this.setState({
       isRefreshing: false,
+      currentActivityPage: this.props.currentActivityPage,
     });
   }
 
+  componentWillReceiveProps(nextProps) {
+    this.state = {
```
Comment (ace0625) [inline src/components/Activity/Activity.js]: Should be done with setState

File: src/components/Chat/ChatPage.js [hunk 2]
```diff
@@ -123,53 +162,54 @@ class ChatPage extends Component {
...
-                }
-              }.bind(this));
           }
-        }.bind(this));
+        }, 1500);
+      }
+    }
+  }
+
+  onReadReceiptUpdated(channel) {
```
Comment (MechanicKim) [inline src/components/Chat/ChatPage.js:187]: empty body!
Is this method necessary?

File: src/components/Main.js [hunk 3]
```diff
@@ -17,72 +21,179 @@ import Text from './Shared/UniText';
...
+        //Sendbird has an issue in updateCurrentUserInfo() API right after connect() API.
+        setTimeout(() => {
+          this.sb.updateCurrentUserInfo(this.props.me.name, this.props.me.profile_picture);
+        }, 1000);
+      }
+
+      NetInfo.isConnected.addEventListener('change', this.onConnectionStateChange.bind(this));
+
+      this.notificationUnsubscribe = Fcm.on('notification', this.onNotificationReceived.bind(this));
+      Fcm.getInitialNotification().then(notif => {
```
Comment (MechanicKim) [inline src/components/Main.js]: If you use '{', '}', it is good to define a function like this.
() => {}
Use '(', ')'
this is airbnb style guide.

File: src/components/Chat/ChatPage.js (hunk 2 above)
Comment (lyw0149) [reply src/components/Chat/ChatPage.js:187]: This will be added when update 'read' feature.

File: src/components/Main.js (hunk 3 above)
Comment (lyw0149) [reply src/components/Main.js]: good!!! thanks!!

File: src/components/Chat/ChatPage.js [hunk 4]
```diff
@@ -123,53 +162,54 @@ class ChatPage extends Component {
...
-        channel: null,
-      });
+  onMessageReceived(channel, userMessage) {
 
-      UserUtil.getOthersProfile(this.onOpponentInfoRequest.bind(this), opponent.userId);
-      const userIds = [me.userId, opponent.userId];
+    if (channel.url == this.state.channel.url) {
+      this.appendSendBirdMessage(userMessage);
+      channel.markAsRead();
+    }else {
```
Comment (SarahhhC) [inline src/components/Chat/ChatPage.js]: add blank.

File: src/components/Chat/react-native-gifted-chat/src/Bubble.js [hunk 5]
```diff
@@ -117,17 +118,33 @@ export default class Bubble extends React.Component {
...
+        <View style = {{ flexDirection: 'row' }}>
+          {this.renderIfRight()}
         <View style={[
           styles[
           this.props.position].wrapper,
           this.props.wrapperStyle[this.props.position],
-          this.handleBubbleToNext(),
-          this.handleBubbleToPrevious(),
+
+          //this.handleBubbleToNext(),
```
Comment (SarahhhC) [inline src/components/Chat/react-native-gifted-chat/src/Bubble.js]: Why hided??

File: src/components/Chat/react-native-gifted-chat/src/Bubble.js [hunk 6]
```diff
@@ -117,17 +118,33 @@ export default class Bubble extends React.Component {
...
+          //this.handleBubbleToPrevious(),
```
Comment (SarahhhC) [inline src/components/Chat/react-native-gifted-chat/src/Bubble.js]: Why hided??

File: src/components/Chat/ChatPage.js (hunk 4 above)
Comment (lyw0149) [reply src/components/Chat/ChatPage.js]: thanks

File: src/components/Chat/react-native-gifted-chat/src/Bubble.js (hunk 6 above)
Comment (lyw0149) [reply src/components/Chat/react-native-gifted-chat/src/Bubble.js]: thanks.

**Comment by MechanicKim** [comment]
@ifndefdeadmau5 Please review this PR!

**Comment by MechanicKim** [comment]
I think this PR is important!
If this PR is passed, local notification problem will be resolved! :-)
I reviewed it but This PR needs to review more.
Did anybody review this? No problem?

File: src/components/App.js [hunk 7]
```diff
@@ -2,6 +2,7 @@ import React, { Component } from 'react';
 import { Dimensions, Platform } from 'react-native';
 import {
   Actions,
+  AppState,
```
Comment (heartjigi) [inline src/components/App.js]: Why do you add this?
I can't find this component at react-native-router-flux.
If you want to use this, you have to move to react-native.

File: src/components/Chat/ChannelList.js [hunk 8]
```diff
@@ -31,79 +32,81 @@ class ChannelList extends Component {
...
-      }
-    } else {
+  connectSendBird(callback) {
+    this.sb.connect(this.props.me._id, (user, error) => {
       this.sb.removeChannelHandler('ChannelList');
-      SendBird().disconnect();
-    }
+      this.sb.addChannelHandler('ChannelList', this.ChannelHandler);
+
+      if (callback) {
```
Comment (heartjigi) [inline src/components/Chat/ChannelList.js]: I think that    
" if (typeof callback === 'function') "
is better

File: src/components/Chat/ChatPage.js [hunk 9]
```diff
@@ -1,17 +1,21 @@
 import React, { Component } from 'react';
 import {
+  AppState,
   AsyncStorage,
```
Comment (MechanicKim) [inline src/components/Chat/ChatPage.js]: 'AsyncStorage' is defined but never used
'ListView' is defined but never used
'Platform' is defined but never used
'RefreshControl' is defined but never used
'ScrollView' is defined but never used
'TextInput' is defined but never used
'TouchableHighlight' is defined but never used

File: src/components/Chat/ChatPage.js [hunk 10]
```diff
@@ -1,17 +1,21 @@
...
   RefreshControl,
   ScrollView,
   StyleSheet,
   TextInput,
   TouchableHighlight,
+  Vibration,
   View,
 } from 'react-native';
 import { Actions } from 'react-native-router-flux';
 import { GiftedChat }  from './react-native-gifted-chat';
```
Comment (MechanicKim) [inline src/components/Chat/ChatPage.js]: Multiple spaces found before 'from'

File: src/components/Chat/ChatPage.js [hunk 11]
```diff
@@ -212,11 +252,12 @@ class ChatPage extends Component {
         onSend={this.onSend.bind(this)}
         channel={this.state.channel}
         user={{
-          _id: this.props.me.userId,
+          _id: this.me.userId,
         }}
         opponentInfo={this.state.opponentInfo}
         loadEarlier={true}
```
Comment (MechanicKim) [inline src/components/Chat/ChatPage.js]: Value must be omitted for boolean attributes

File: src/components/Chat/ChatPage.js [hunk 12]
```diff
@@ -212,11 +252,12 @@ class ChatPage extends Component {
...
         renderFooter={this.renderFooter}
+        isLoadingEarlier={true}
```
Comment (MechanicKim) [inline src/components/Chat/ChatPage.js]: Value must be omitted for boolean attributes

File: src/components/Chat/ChatPage.js (hunk 10 above)
Comment (lyw0149) [reply src/components/Chat/ChatPage.js]: eagle eye!!

File: src/components/Chat/ChannelList.js (hunk 8 above)
Comment (lyw0149) [reply src/components/Chat/ChannelList.js]: right.

File: src/components/App.js (hunk 7 above)
Comment (lyw0149) [reply src/components/App.js]: My mistake.

File: src/components/Chat/ChatPage.js (hunk 11 above)
Comment (lyw0149) [reply src/components/Chat/ChatPage.js]: Thank you for your information.

File: src/components/Chat/ChatPage.js (hunk 12 above)
Comment (lyw0149) [reply src/components/Chat/ChatPage.js]: Thank you for your information.

File: src/components/Chat/ChatPage.js (hunk 11 above)
Comment (lyw0149) [reply src/components/Chat/ChatPage.js]: But it must? or it can?

**Comment by ifndefdeadmau5** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/sv-bootcamp/bridgeme-client/pull-request/108/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=sv-bootcamp/bridgeme-client)

**Pull request merged**